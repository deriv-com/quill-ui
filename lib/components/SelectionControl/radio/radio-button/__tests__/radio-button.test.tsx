import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import RadioButton from "..";

describe("RadioButton component", () => {
    it("should render correctly with default props", () => {
        const { container } = render(<RadioButton>Radio Option</RadioButton>);
        const label = screen.getByText(/Radio Option/i);
        expect(label).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should render with a custom className", () => {
        const { container } = render(
            <RadioButton className="custom-class">Radio Option</RadioButton>,
        );
        const input = screen.getByTestId("dt_quill_radio_button");
        expect(input).toBeInTheDocument();
        expect(input.classList.contains("custom-class")).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

    it("should be checked by default if defaultChecked is true", () => {
        const { container } = render(
            <RadioButton defaultChecked id="test-id" value="option1">
                Radio Option
            </RadioButton>,
        );
        expect(
            screen.getByTestId("dt_checked_icon_option1_false"),
        ).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should not be checked by default if defaultChecked is false", () => {
        const { container } = render(
            <RadioButton id="test-id" value="option2">
                Radio Option
            </RadioButton>,
        );
        expect(
            screen.getByTestId("dt_unchecked_icon_option2_false"),
        ).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should be disabled if disabled is true", () => {
        const { container } = render(
            <RadioButton disabled id="test-id" value="option3">
                Radio Option
            </RadioButton>,
        );
        expect(
            screen.getByTestId("dt_unchecked_icon_option3_true"),
        ).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should render info icon when hasInfo is true", () => {
        const { container } = render(
            <RadioButton hasInfo classNameInfo="custom-info">
                Radio Option
            </RadioButton>,
        );
        const infoIcon = screen.getByTestId("dt_quill_radio_button_info");
        expect(infoIcon.classList.contains("custom-info")).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

    it("should not render info icon when hasInfo is false", () => {
        const { container } = render(<RadioButton>Radio Option</RadioButton>);
        const infoIcon = screen.queryByTestId("dt_quill_radio_button_info");
        expect(infoIcon).toBeNull();
        expect(container).toMatchSnapshot();
    });

    it("should not trigger click event on input when info icon is clicked and disabled", () => {
        const mockOnChange = jest.fn();
        const { container } = render(
            <RadioButton
                disabled
                id="test-id"
                onChange={mockOnChange}
                value="option4"
            >
                Radio Option
            </RadioButton>,
        );
        const input = screen.getByTestId("dt_unchecked_icon_option4_true");
        userEvent.click(input);
        expect(mockOnChange).toHaveBeenCalledTimes(0);
        expect(container).toMatchSnapshot();
    });
});
