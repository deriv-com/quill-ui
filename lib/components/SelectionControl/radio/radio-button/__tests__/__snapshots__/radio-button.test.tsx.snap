// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component should be checked by default if defaultChecked is true 1`] = `
<div>
  <label
    class="quill-radio-button quill-radio-group__item--active"
    data-testid="dt_quill_radio_button"
    for="test-id"
  >
    <input
      checked=""
      class="quill-radio-button__input"
      type="radio"
      value="option1"
    />
    <span
      class="quill-radio-button__icon quill-radio-button__icon--left"
      id="test-id"
    >
      <svg
        data-testid="dt_checked_icon_option1_false"
        fill="#000000"
        height="32"
        role="img"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 26.5c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m0-13.75c-1.367 0-2.578.742-3.281 1.875-.664 1.172-.664 2.617 0 3.75.703 1.172 1.914 1.875 3.281 1.875 1.328 0 2.54-.703 3.242-1.875.664-1.133.664-2.578 0-3.75-.703-1.133-1.914-1.875-3.242-1.875"
        />
      </svg>
    </span>
    <span
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-radio-button__label"
    >
      Radio Option
    </span>
  </label>
</div>
`;

exports[`RadioButton component should be disabled if disabled is true 1`] = `
<div>
  <label
    class="quill-radio-button"
    data-testid="dt_quill_radio_button"
    for="test-id"
  >
    <input
      class="quill-radio-button__input"
      disabled=""
      type="radio"
      value="option3"
    />
    <span
      class="quill-radio-button__icon quill-radio-button__icon--disabled quill-radio-button__icon--left"
      id="test-id"
    >
      <svg
        data-testid="dt_unchecked_icon_option3_true"
        fill="#b8b8b8"
        height="32"
        role="img"
        tabindex="0"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5"
        />
      </svg>
    </span>
    <span
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-radio-button__label quill-radio-button__label--disabled"
    >
      Radio Option
    </span>
  </label>
</div>
`;

exports[`RadioButton component should not be checked by default if defaultChecked is false 1`] = `
<div>
  <label
    class="quill-radio-button"
    data-testid="dt_quill_radio_button"
    for="test-id"
  >
    <input
      class="quill-radio-button__input"
      type="radio"
      value="option2"
    />
    <span
      class="quill-radio-button__icon quill-radio-button__icon--left"
      id="test-id"
    >
      <svg
        data-testid="dt_unchecked_icon_option2_false"
        fill="#7e7e7e"
        height="32"
        role="img"
        tabindex="0"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5"
        />
      </svg>
    </span>
    <span
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-radio-button__label"
    >
      Radio Option
    </span>
  </label>
</div>
`;

exports[`RadioButton component should not render info icon when hasInfo is false 1`] = `
<div>
  <label
    class="quill-radio-button"
    data-testid="dt_quill_radio_button"
  >
    <input
      class="quill-radio-button__input"
      type="radio"
      value=""
    />
    <span
      class="quill-radio-button__icon quill-radio-button__icon--left"
    >
      <svg
        data-testid="dt_unchecked_icon_undefined_false"
        fill="#7e7e7e"
        height="32"
        role="img"
        tabindex="0"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5"
        />
      </svg>
    </span>
    <span
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-radio-button__label"
    >
      Radio Option
    </span>
  </label>
</div>
`;

exports[`RadioButton component should not trigger click event on input when info icon is clicked and disabled 1`] = `
<div>
  <label
    class="quill-radio-button"
    data-testid="dt_quill_radio_button"
    for="test-id"
  >
    <input
      class="quill-radio-button__input"
      disabled=""
      type="radio"
      value="option4"
    />
    <span
      class="quill-radio-button__icon quill-radio-button__icon--disabled quill-radio-button__icon--left"
      id="test-id"
    >
      <svg
        data-testid="dt_unchecked_icon_option4_true"
        fill="#b8b8b8"
        height="32"
        role="img"
        tabindex="0"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5"
        />
      </svg>
    </span>
    <span
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-radio-button__label quill-radio-button__label--disabled"
    >
      Radio Option
    </span>
  </label>
</div>
`;

exports[`RadioButton component should render correctly with default props 1`] = `
<div>
  <label
    class="quill-radio-button"
    data-testid="dt_quill_radio_button"
  >
    <input
      class="quill-radio-button__input"
      type="radio"
      value=""
    />
    <span
      class="quill-radio-button__icon quill-radio-button__icon--left"
    >
      <svg
        data-testid="dt_unchecked_icon_undefined_false"
        fill="#7e7e7e"
        height="32"
        role="img"
        tabindex="0"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5"
        />
      </svg>
    </span>
    <span
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-radio-button__label"
    >
      Radio Option
    </span>
  </label>
</div>
`;

exports[`RadioButton component should render info icon when hasInfo is true 1`] = `
<div>
  <label
    class="quill-radio-button"
    data-testid="dt_quill_radio_button"
  >
    <input
      class="quill-radio-button__input"
      type="radio"
      value=""
    />
    <span
      class="quill-radio-button__icon quill-radio-button__icon--left"
    >
      <svg
        data-testid="dt_unchecked_icon_undefined_false"
        fill="#7e7e7e"
        height="32"
        role="img"
        tabindex="0"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5"
        />
      </svg>
    </span>
    <span
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-radio-button__label"
    >
      Radio Option
    </span>
    <svg
      class="quill-radio-button__info custom-info"
      data-testid="dt_quill_radio_button_info"
      height="24"
      role="img"
      viewBox="0 0 16 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M8 5a6.98 6.98 0 0 0-6.062 3.5c-1.282 2.188-1.282 4.844 0 7C3.188 17.688 5.467 19 8 19c2.5 0 4.781-1.312 6.031-3.5 1.281-2.156 1.281-4.812 0-7C12.781 6.344 10.5 5 8 5m0 15c-2.875 0-5.5-1.5-6.937-4-1.438-2.469-1.438-5.5 0-8A8.02 8.02 0 0 1 8 4c2.844 0 5.469 1.531 6.906 4 1.438 2.5 1.438 5.531 0 8A7.94 7.94 0 0 1 8 20m-1.5-5h1v-3h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5H8c.25 0 .5.25.5.5V15h1c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-3a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5M8 9.75A.72.72 0 0 1 7.25 9 .74.74 0 0 1 8 8.25a.76.76 0 0 1 .75.75.74.74 0 0 1-.75.75"
        />
      </g>
      <defs>
        <clippath
          id="b0a404553cdcc487345364cc56169aef__a"
        >
          <path
            d="M0 0h16v24H0z"
          />
        </clippath>
      </defs>
    </svg>
  </label>
</div>
`;

exports[`RadioButton component should render with a custom className 1`] = `
<div>
  <label
    class="quill-radio-button custom-class"
    data-testid="dt_quill_radio_button"
  >
    <input
      class="quill-radio-button__input"
      type="radio"
      value=""
    />
    <span
      class="quill-radio-button__icon quill-radio-button__icon--left"
    >
      <svg
        data-testid="dt_unchecked_icon_undefined_false"
        fill="#7e7e7e"
        height="32"
        role="img"
        tabindex="0"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5"
        />
      </svg>
    </span>
    <span
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-radio-button__label"
    >
      Radio Option
    </span>
  </label>
</div>
`;
